static/js/notes.js

```diff
@@ -0,0 +1,55 @@
+var USERS_BY_ID = {};
+
+const $createAuthor = function(authorId) {
+    const user = USERS_BY_ID[authorId];
+
+    return $('<span>', {text: user.displayName});
+};
+
+const $createType = function(type) {
+    return $('<span>', {text: type});
+};
+
+const $createCreationTime = function(creationTime) {
+    return $('<span>', {text: moment(creationTime).fromNow()});
+};
+
+const $createContent = function(content) {
+    return $('<div>', {text: content});
+};
+
+const $createDivider = function() {
+    return $('<span>', {text: ' - '});
+};
+
+const $createNote = function(note) {
+    const $author = $createAuthor(note.authorId);
+    const $type = $createType(note.type);
+    const $creationTime = $createCreationTime(note.creationTime);
+    const $content = $createContent(note.content);
+
+    const $header = $('<div>').addClass('note-header')
+        .append($author).append($createDivider())
+        .append($type).append($createDivider())
+        .append($creationTime);
+
+    return $('<div>').addClass('note')
+        .append($header)
+        .append($content);
+};
+
+$(function() {
+    $.get('/api/user', function(usersById) {
+        USERS_BY_ID = usersById;
```

Comment:
If i understand correctly, we need this because the note object stores a userId and not a user Display Name. So we need to convert userId's to User?

And normally we'd convert that on the database side, however, that could lead to mistakes if different users had the same display name?

Reply:
database side? i assume you just mean server side? We could certainly change the notes api to get the names and fill them in before giving it to the client side, but that seemed like it was mucking up the purity of the notes api.

Reply:
I agree, but i'm not sure why. Would you mind explaining how it mucks with the purity of the notes api?